### PR TITLE
refactor: use guard getters

### PR DESCRIPTION
**Pull request by erights**
To pave the way for https://github.com/endojs/endo/pull/1712 , switch from directly accessing the fields of guards to using the guard type's validating payload getter, and then accessing the fields of the payload.

See also https://github.com/endojs/endo/pull/1771, which would enable us to remove the two `@ts-expect-error`s that we had to add.

### CODE REVIEW

**Comment by erights** [comment]
See https://github.com/Agoric/agoric-sdk/pull/8340

File: packages/zoe/src/typeGuards.js [hunk 1]
```diff
@@ -224,7 +224,7 @@ export const InstanceStorageManagerIKit = harden({
       InstanceHandleShape,
       M.remotable('instanceAdmin'),
     ).returns(M.promise()),
-    deleteInstanceAdmin: M.call(InstanceAdminI).returns(),
+    deleteInstanceAdmin: M.call(InstanceAdminShape).returns(),
```
Comment (erights) [inline packages/zoe/src/typeGuards.js:227]: This bug was only caught by seeing https://github.com/Agoric/agoric-sdk/pull/8340 fail , because it uses the stricter https://github.com/endojs/endo/pull/1712

Comment (erights) [reply packages/zoe/src/typeGuards.js:227]: See https://github.com/Agoric/agoric-sdk/issues/8341

File: packages/internal/src/callback.js [hunk 2]
```diff
@@ -315,7 +316,7 @@ harden(prepareAttenuator);
  */
 export const prepareGuardedAttenuator = (zone, interfaceGuard, opts = {}) => {
   // TODO: Handle symbolMethodGuards too?
-  const { methodGuards } = interfaceGuard;
+  const { methodGuards } = getInterfaceGuardPayload(interfaceGuard);
   const methodNames = ownKeys(methodGuards);
```
Comment (michaelfig) [inline packages/internal/src/callback.js]: As an aside, would you consider implementing this TODO as a drive-by (or at least enlighten me regarding the best way to get all the symbol and string method names for an `interfaceGuard`)?

**Review by michaelfig — APPROVED** [review]
LGTM!  I'd love to see this land in the SDK.

File: packages/internal/src/callback.js (hunk 2 above)
Comment (erights) [reply packages/internal/src/callback.js]: Drive by done.

File: packages/internal/src/callback.js [hunk 3]
```diff
@@ -314,9 +315,7 @@ harden(prepareAttenuator);
  * @param {string} [opts.tag]
  */
 export const prepareGuardedAttenuator = (zone, interfaceGuard, opts = {}) => {
-  // TODO: Handle symbolMethodGuards too?
-  const { methodGuards } = interfaceGuard;
-  const methodNames = ownKeys(methodGuards);
+  const methodNames = getInterfaceMethodKeys(interfaceGuard);
```
Comment (michaelfig) [inline packages/internal/src/callback.js:318]: 💯
